yt_dlp/extractor/mxplayer.py

```diff
@@ -106,73 +106,93 @@ class MxplayerIE(InfoExtractor):
-        API_URL = 'https://androidapi.mxplay.com/v1/detail/'
-        headers = {
-            'X-Av-Code': '23',
-            'X-Country': 'IN',
-            'X-Platform': 'android',
-            'X-App-Version': '1370001318',
-            'X-Resolution': '3840x2160',
-        }
-        data_json = self._download_json(f'{API_URL}{type}/{video_id}', display_id, headers=headers)['profile']
-
-        season, series = None, None
-        for dct in data_json.get('levelInfos', []):
-            if dct.get('type') == 'tvshow_season':
-                season = dct.get('name')
-            elif dct.get('type') == 'tvshow_show':
-                series = dct.get('name')
-        thumbnails = []
-        for thumb in data_json.get('poster', []):
-            thumbnails.append({
-                'url': thumb.get('url'),
-                'width': thumb.get('width'),
-                'height': thumb.get('height'),
-            })
+        video_type, display_id, video_id = self._match_valid_url(url).groups()
+
+        if 'show' in video_type:
+            video_type = 'episode'
+
+        data_json = self._download_json(
+            f'https://api.mxplay.com/v1/web/detail/video?type={video_type}&id={video_id}', display_id)
 
         formats = []
         subtitles = {}
-        for dct in data_json.get('playInfo', []):
-            if dct.get('extension') == 'mpd':
-                frmt, subs = self._extract_mpd_formats_and_subtitles(dct.get('playUrl'), display_id, fatal=False)
-                formats.extend(frmt)
-                subtitles = self._merge_subtitles(subtitles, subs)
-            elif dct.get('extension') == 'm3u8':
-                frmt, subs = self._extract_m3u8_formats_and_subtitles(dct.get('playUrl'), display_id, fatal=False)
-                formats.extend(frmt)
-                subtitles = self._merge_subtitles(subtitles, subs)
+        series, season, season_number, episode_number = None, None, None, None
+
+        if video_type == 'episode':
+            series = traverse_obj(data_json, ('container', 'container', 'title'))
+            season = traverse_obj(data_json, ('container', 'title'))
+            episode_number = data_json.get('sequence')
+            season_number = int_or_none(self._search_regex(r'Season (\d+)', season, 'Season Number', fatal=False))
+
+        for stream_type in 'dash', 'hls':
+            playlist_url = f'https://llvod.mxplay.com/{data_json["stream"][stream_type]["high"]}'
+
+            if stream_type == 'dash':
+                frmts, subs = self._extract_mpd_formats_and_subtitles(playlist_url, display_id, fatal=False)
+            else:
+                frmts, subs = self._extract_m3u8_formats_and_subtitles(playlist_url, display_id, fatal=False)
+
+            formats.extend(frmts)
+            self._merge_subtitles(subs, target=subtitles)
+
         self._sort_formats(formats)
+
         return {
             'id': video_id,
+            'title': data_json['title'],
```

Comment:
Title shouldn't fatal
```suggestion
            'title': data_json.get('title'),
```

Reply:
'title' field is mandatory AFAIK. So I believe it doesn't make sense to make it non-fatal.

Reply:
https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#mandatory-and-optional-metafields
> While all extractors must return a `title`, they must also allow it's extraction to be non-fatal

Reply:
Oh. My bad for not reading it fully. Done.